src/main/java/io/getunleash/repository/FeatureRepository.java

```diff
@@ -0,0 +1,100 @@
+package io.getunleash.repository;
+
+import io.getunleash.FeatureToggle;
+import io.getunleash.Segment;
+import io.getunleash.UnleashException;
+import io.getunleash.event.EventDispatcher;
+import io.getunleash.event.UnleashReady;
+import io.getunleash.lang.Nullable;
+import io.getunleash.util.UnleashConfig;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FeatureRepository implements IFeatureRepository {
+    private final FeatureBackupHandlerFile featureBackupHandler;
+    private final FeatureBootstrapHandler featureBootstrapHandler;
+    private final HttpFeatureFetcher featureFetcher;
+    private final EventDispatcher eventDispatcher;
+
+    private FeatureCollection featureCollection;
+    private boolean ready;
+
+    private static FeatureRepository instance = null;
+
+    private FeatureRepository(UnleashConfig unleashConfig) {
```

Comment:
Why the static `FeatureRepository` approach with private constructor? It forbids an app from pointing to 2 Unleash servers (admittedly an unusual use case, but not unreasonable)

The static instances being set below could just as easily be constructor arguments, which simplify the unit tests and just keep things from being fragile.